sihl/src/database_migration_repo.ml

```diff
@@ -23,79 +23,91 @@ end
 module type Sig = sig
   module Migration = Migration
 
-  val create_table_if_not_exists : unit -> unit Lwt.t
-  val get : namespace:string -> Migration.t option Lwt.t
-  val get_all : unit -> Migration.t list Lwt.t
-  val upsert : state:Migration.t -> unit Lwt.t
+  val create_table_if_not_exists : string -> unit Lwt.t
+  val get : string -> namespace:string -> Migration.t option Lwt.t
+  val get_all : string -> Migration.t list Lwt.t
+  val upsert : string -> Migration.t -> unit Lwt.t
 end
 
+(* Common functions *)
+let get_request table =
+  Caqti_request.find_opt
+    Caqti_type.string
+    Caqti_type.(tup3 string int bool)
+    (Format.sprintf
+       {sql|
+       SELECT
+         namespace,
+         version,
+         dirty
+       FROM %s
```

Comment:
Could this be an attack vector for SQL injection?

Reply:
No, the table name is not based on any end user input. You need access to the env variables to set it.